backend/src/utils/agent.v1.py

```
@@ -0,0 +1,226 @@
+        self.checkpointer = checkpointer
+        system = self.config.get('configurable').get("system", None)
+        # Get MCP tools if provided
+        if mcp and len(mcp.keys()) > 0:
+            mcp_service = McpService(mcp)
+            self.tools.extend(await mcp_service.get_tools())
+            
+        if a2a and len(a2a.keys()) > 0:
+            # Check if a2a is a dictionary with multiple entries
+            if isinstance(a2a, dict):
+                # Loop through each entry in the a2a dictionary
+                for key, config in a2a.items():
```

Ruff (0.11.9)

Loop control variable `key` not used within loop body

Rename unused `key` to `_key`

(B007)